tree/20_cfe_basics/cfengine/packages.cf

```diff
@@ -99,6 +99,19 @@ bundle common rpm_knowledge
       "rpm3_arch_regex" string => "\S+\s+(\S+).*";
 }
 
+bundle common redhat_no_locking_knowledge {
+  vars:
+    !redhat_setopt_not_available::
+      "no_locking_command" string => "--setopt=exit_on_lock=True";
+    redhat_setopt_not_available::
+      "no_locking_command" string => "";
+
+
+  classes:
+    redhat::
+      "redhat_setopt_not_available" expression => returnszero("LANG=C ${paths.path[yum]} --setopt=exit_on_lock=True 2>&1 | ${paths.path[grep]} -E '(Command line error: no such option: --setopt|Main config did not have a exit_on_lock attr. before setopt)'", "useshell");
```

Comment:
this option is only used in conjonction with yum binary, and checked on redhat only.
I agree that reversing the condition will make it clearer